packages/cli/src/cmds/agentInstaller/projectConfiguration.ts

```diff
@@ -81,12 +87,103 @@ async function resolveSelectedInstallers(
+// error Command "workspaces" not found.
+async function getYarnSubprojectsVersionOne(
+  dir: string,
+  installer: YarnInstaller,
+  subprojects: ProjectConfiguration[]
+) {
+  const packageJsonFilename = join(installer.path, 'package.json');
+  try {
+    const data = await fs.readFile(packageJsonFilename, 'utf-8');
+    const packageJson = JSON.parse(data);
+    if ('workspaces' in packageJson) {
+      const workspaces = packageJson['workspaces'];
+      // process as glob because it could be something like
+      // workspaces: [ 'packages/*' ]
+      for (const workspace of workspaces) {
+        const matches = glob.sync(workspace, {
+          matchBase: true,
+          cwd: installer.path,
+          ignore: ['**/node_modules/**', '**/.git/**'],
+          strict: false,
+        });
+        for (const yarnPackage of matches) {
+          // packages must be directories; don't break on leftover files
+          const isDir = lstatSync(join(installer.path, yarnPackage)).isDirectory();
+          if (isDir) {
+            const subProjectPath = join(dir, yarnPackage);
+            const subProjectName = basename(yarnPackage);
+            subprojects.push({
+              name: subProjectName,
+              path: subProjectPath,
+              // must set the new path in the installer
+              availableInstallers: [new YarnInstaller(subProjectPath)],
+            } as ProjectConfiguration);
+          }
+        }
+      }
+    }
+  } catch (err) {
+    if ((err as NodeJS.ErrnoException).code === 'ENOENT') {
+      // there's no package.json file. don't crash
+    } else {
+      throw err;
+    }
+  }
+}
+
+async function getYarnSubprojectsVersionAboveOne(
+  dir: string,
+  installer: YarnInstaller,
+  subprojects: ProjectConfiguration[]
+) {
+  const cmd = new CommandStruct('yarn', ['workspaces', 'list', '--json'], installer.path);
+  const output = await run(cmd);
+
+  for (const line of output.stdout.split('\n')) {
+    if (line !== '') {
+      const yarnWorkspace = JSON.parse(line);
+      const subProjectPath = join(dir, yarnWorkspace['location']);
+      const subProjectName = basename(yarnWorkspace['name']);
+      subprojects.push({
+        name: subProjectName,
+        path: subProjectPath,
+        // must set the new path in the installer
+        availableInstallers: [new YarnInstaller(subProjectPath)],
+      } as ProjectConfiguration);
+    }
+  }
+}
+
+export async function getYarnSubprojects(
+  dir: string,
+  availableInstallers: AgentInstaller[]
+): Promise<ProjectConfiguration[]> {
+  let subprojects: ProjectConfiguration[] = [];
+
+  for (const installer of availableInstallers) {
+    if (installer.name === 'yarn') {
+      if (await (installer as YarnInstaller).isYarnVersionOne()) {
+        await getYarnSubprojectsVersionOne(dir, installer as YarnInstaller, subprojects);
+      } else {
+        await getYarnSubprojectsVersionAboveOne(dir, installer as YarnInstaller, subprojects);
```

Comment:
Declared a local variable that coerces the type and used this variable instead of repeated coercion. 
```
      const yarnInstaller = installer as YarnInstaller;
```

Reply:
A type check would have been cleaner, but ok :)